hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/RepairingTask.java

```diff
@@ -166,69 +162,54 @@ private void scheduleNextRun() {
     }
 
     public <K, V> RepairingHandler registerAndGetHandler(String name, NearCache<K, V> nearCache) {
-        boolean started = running.compareAndSet(false, true);
-        if (started) {
-            assignAndGetUuids();
-        }
-
-        RepairingHandler repairingHandler = handlers.get(name);
-        if (repairingHandler == null) {
-            repairingHandler = new RepairingHandler(logger, localUuid, name, nearCache, serializationService, partitionService);
-            repairingHandler.initUnknownUuids(partitionUuids);
-
-            StaleReadDetector staleReadDetector = new StaleReadDetectorImpl(repairingHandler, partitionService);
+        RepairingHandler handler = handlers.get(name);
+        if (handler == null) {
```

Comment:
Hmm, the `running.compareAndSet(false, true)` part looks like this method has to be thread-safe. But the creation of the `ReparingHandler` seems not to be (check -> create -> put). Is there are risk of creating two `RepairingHandler` instances? Or is the `AtomicBoolean` just used for the convenience of the `compareAndSet()` method?